src/compilation_context.ml

```diff
@@ -120,7 +120,11 @@ let create ~super_context ~scope ~expander ~obj_dir
   }
 
 let for_alias_module t =
-  let flags = Ocaml_flags.default ~profile:(SC.profile t.super_context) in
+  let flags =
+    let project = Scope.project t.scope in
+    let dune_version = Dune_project.dune_version project in
+    Ocaml_flags.default ~profile:(SC.profile t.super_context) ~dune_version
```

Comment:
Would it be worth extracting those three lines into a:
```ocaml
val defaut_ocaml_flags : scope: Scope.t -> sctx: Super_context.t -> Ocaml_flags.t
```
somewhere ?

Reply:
Previously we'd have plenty of such functions, but eventually Jeremie and I decided they're not worth the trouble. All of these helper functions that just chain projections and have no logic ended up really bloating our API. So I think that as a rule of a thumb, we try to do without them.

Though I agree that with this case it's quite tempting. Maybe we should just have `Ocaml_flags.default_flags` just take scope or the project?

Reply:
> Maybe we should just have Ocaml_flags.default_flags just take scope or the project?

This cannot work as it introduced cycles.